sdk/python/kfp/compiler/test_data/components/container_with_if_placeholder.py

```diff
@@ -16,18 +16,25 @@
 from kfp.components import placeholders
 from kfp.dsl import container_component
 from kfp.dsl import ContainerSpec
+from kfp.dsl import Dataset
+from kfp.dsl import Output
 from kfp.dsl import OutputPath
 
 
 @container_component
-def container_with_if_placeholder(optional_input: Optional[str],
-                                  output_path: OutputPath(str)):
+def container_with_if_placeholder(output_path: OutputPath(str),
+                                  dataset: Output[Dataset],
+                                  optional_input: str = 'default'):
     return ContainerSpec(
         image='python:3.7',
         command=[
             'my_program',
             placeholders.IfPresentPlaceholder(
-                input_name='optional_input', then='hi', else_='bye')
+                input_name='optional_input',
+                then=[optional_input],
```

Comment:
Probably out of scope for this PR. Do you think if we can make the square brackets optional? i.e. `then=optional_input` is equivalent to `then=[optional_input]`?

Reply:
Yes, I think it's a reasonable change. I will submit a new PR to fix this issue.